File: DesignPatternsDotNet/ChainOfResponsibility/MyHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ChainOfResponsibility
{
    public abstract class MyHandler
    {
        protected MyHandler myNextHandler;

        public MyHandler NextHandler
        {       
            set
            {
            myNextHandler = value;
            }
        }
        public abstract void DoThis(Task task);
       

    }
}

File: DesignPatternsJava/src/main/java/com/ChainOfResponsibility/MyHandler.java
package com.ChainOfResponsibility;

public abstract class MyHandler {

	protected MyHandler myNextHandler;
	/***
	 * to pass the request to successor, this will be used for making chain
	 * @param successor
	 */
	public void setMyNextHandler(MyHandler successor)
	{
		myNextHandler = successor;
	}
	/***
	 * This method actually responsible for doing task
	 * @param task
	 */
	public abstract void doThis(Task task);	

}

	
	